### ISSUE TITLE
SABnzbd applying permissions to mounted volumes (incomplete-downloads and downloads)

### ISSUE THREAD

**Issue by jpdsc** [linuxserver/docker-sabnzbd#93]
[linuxserverurl]: https://linuxserver.io
[![linuxserver.io](https://raw.githubusercontent.com/linuxserver/docker-templates/master/linuxserver.io/img/linuxserver_medium.png)][linuxserverurl]

------------------------------

## Expected Behavior
When assigning a Temporary Download Folder and Completed Download Folder in SABnzbd, the original permissions should stay intact.

## Current Behavior
Whenever the container is created, SABnzbd applies permissions 755 to the folder set as Temporary Download Folder and Completed Download Folder and therefore overwriting any other permissions added to the folder.
I have cross-checked this with other download clients and none of them is doing this to the folders.

## Steps to Reproduce
1. Create Temporary Download Folder and Completed Download Folder 
2. Change the permissions to something different than 755
3. Run the container with the volumes mounted
4. Check the permissions `ls -ld `
5. The permission is now changed from the original permission to 755

## Environment
**OS:** Synology
**CPU:** architecture:** x86_64
**How docker service was installed:**
Synology package manager

## Command used to create docker container (run/create/compose/screenshot)
docker-compose:

```
    sabnzbd:
        container_name: SABnzbd
        environment:
            - TZ=Europe/Amsterdam
            - PGID=101
            - PUID=1026
        volumes:
            - '/volume2/docker/SABnzbd/config:/config:rw'
            - '/volume1/Shared:/shared' ## in the SABnzbd GUI I set the correct path
        networks:
            private:
                ipv4_address: 10.2.0.2
        hostname: NAS
        ports:
            - '8080:8080'
            - '9090:9090'
        restart: unless-stopped
        image: 'linuxserver/sabnzbd:latest'
```
## Docker logs
```
joao@NAS:/volume2/docker/Compose/Production$ ls -ld "/volume1/Shared/Downloading/"
drwxr-xr-x 1 joao administrators 0 Dec 23 10:51 /volume1/Shared/Downloading/
joao@NAS:/volume2/docker/Compose/Production$ ls -ld "/volume1/Shared/Downloads/Unsorted/"
drwxr-xr-x 1 joao administrators 0 Dec 23 10:32 /volume1/Shared/Downloads/Unsorted/
```

I have a thread with SABnzbd regarding this here: https://forums.sabnzbd.org/viewtopic.php?f=2&t=24696&p=123104#p123104

Is there anything I can do to resolve this? Or is it possible to resolve this through the container creation?
Edit: it seems to also change permission of the config folder to Everyone regardless if you chown it to a different user...

**Comment by aptalca** [comment]
Our image does not chmod, it only chowns, meaning, it doesn't change permission levels, it only changes the owner (sets it to the PUID/PGID you use). It is done on `/config` so that sab can access its own settings and data, and it does it (non-recursively) on the `/downloads` and `/incomplete-downloads` folders, which you are not using anyway.

**Comment by aptalca** [comment]
Oh and also, the volume mapping you're using for your downloads, `/shared` is not even recognized by our image. Therefore we don't touch it at all.

**Comment by jpdsceu** [comment]
> Oh and also, the volume mapping you're using for your downloads, `/shared` is not even recognized by our image. Therefore we don't touch it at all.

Yes I am using different volume mapping to ensure it works with Radarr and Sonarr.
I think the issue is on SABnzbd side and not in the Docker image... SABnzbd applies 755 to the folder /shared/Downloading and /shared/Downloads/Unsorted

I don't really care about the 755 as I can have that changed to 777 in the Sab config. The issue is that SABnzbd chowns access to everyone on the 2 folders and the config which any other client doesn't.

Config folder
![image](https://user-images.githubusercontent.com/7014736/103004970-1b156380-4532-11eb-8baf-700ace3390e4.png)

Downloading folder:
![image](https://user-images.githubusercontent.com/7014736/103005379-36806e80-4532-11eb-877e-271a0e19ac91.png)

So I guess I have to wait for an update from the developer himself as, in my opinion, this is not how the permissions should be handled within a application.

**Comment by AronDurkinSSS** [comment]
did you ever get this resolved @jpdsc ? I'm having the same issue, no idea why

**Comment by jpdsc** [comment]
> did you ever get this resolved @jpdsc ? I'm having the same issue, no idea why

I didn't. I moved to nzbget where I could add the attributes to the unpack to ensure it doesn't overwrite permissions.

**Comment by AronDurkinSSS** [comment]
hmm, a shame, thanks for getting back anyway

**Comment by Zyprexia** [comment]
I found a workaround. I was using UID 1000 and GID 1000, but every time I tried to change the default temp and complete folder I got access denied messages during save. I connected to the console of the container and I was able to access the volumes sabnzbd was unable to. I also was able to touch files and make directories under completed directory. But sabnzbd was still unable to.

Finally I changed the GID/UID of the container to 0 and under root user/group sabnzbd is working as it should. When I use a different UID/GID it cannot access the volumes. 

Running as root is not a best practice, so I have placed quotas and other limitations of the /downloads folder. Most episodes and movies are only watched once and are removed afterwards. If I want to save something I have to manually move it to a permanent location.

Maybe this helps others.
